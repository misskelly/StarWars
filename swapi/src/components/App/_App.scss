* {
  box-sizing: border-box;
}

.App,
.home {
  position: relative;
  width: 100%;
  height: 100vh;
  z-index: 1;
}

.home {
  display: grid;
  grid-template-columns: 2fr 3fr 2fr;
  grid-template-rows: 10rem 3fr 1fr;
  grid-template-areas:
    "header header header"
    "left film right"
    "left ship right";
}




.left-btns {
  grid-area: left;
  display: grid;
  grid-template-areas:
    ". . ."
    "moon . planets"
    ". . ."
    "people people people"
    "people people people"
    "people people people"
}

.moon {
  grid-area: moon;
  border-radius: 100%;
  background: rgb(160, 154, 154);
  height: 2rem;
  width: 2rem;
  margin: 7rem 0 0 3rem;
}

.planets-btn {
  grid-area: planets;
  padding: 2rem;
  height: 15rem;
  width: 15rem;
  // margin: 3rem;
}

.people-btn {
  grid-area: people;
  padding: 2rem;
  height: 11rem;
  width: 11rem;
  margin: 0 0 0 5rem;
}



.right-btns {
  grid-area: right;
  display: grid;
  grid-template-areas:
    "favs favs ."
    "favs favs ."
    ". . ."
    "ships ships ships"
    "ships ships ships"
    "ships ships ships"
    "ships ships ships"
}

.favs-btn {
  grid-area: favs;
  padding: 2rem;
  height: 13rem;
  width: 13rem;
  // margin: 0 0 3rem 5rem;
}

.vehicles-btn {
  grid-area: ships;
  height: 22rem;
  width: 22rem;
  margin: 2rem;
}

.planet-btn {
  -moz-box-shadow: 3px 4px 0px 0px #8a2a21;
  -webkit-box-shadow: 3px 4px 0px 0px #8a2a21;
  box-shadow: 3px 4px 0px 0px #8a2a21;
  background: -webkit-gradient(linear, left top, left bottom, color-stop(0.05, #c62d1f), color-stop(1, #f24437));
  background: -moz-linear-gradient(top, #c62d1f 5%, #f24437 100%);
  background: -webkit-linear-gradient(top, #c62d1f 5%, #f24437 100%);
  background: -o-linear-gradient(top, #c62d1f 5%, #f24437 100%);
  background: -ms-linear-gradient(top, #c62d1f 5%, #f24437 100%);
  background: linear-gradient(to bottom, #c62d1f 5%, #f24437 100%);
  filter: progid:DXImageTransform.Microsoft.gradient(startColorstr='#c62d1f', endColorstr='#f24437', GradientType=0);
  background-color: #c62d1f;
  -moz-border-radius: $circle;
  -webkit-border-radius: $circle;
  border-radius: $circle;
  border: 1px solid #d02718;
  display: inline-block;
  cursor: pointer;
  color: #ffffff;
  font-family: $alien;
  font-size: 2rem;
  text-decoration: none;
  text-shadow: 0px 1px 0px #810e05;
}