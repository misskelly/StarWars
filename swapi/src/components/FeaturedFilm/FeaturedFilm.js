import React from 'react';

const FeaturedFilm = ({film}) => {
  return (
    <section className='film-container'>
      <article className='scroll-container'>
      {film.scroll}
      </article>
      <article className='film-info'>
        <h3 className='film-title'>{film.title}</h3>
        <h4 className='release-year'>{film.year}</h4>
      </article>
    </section>
  )
}

export default FeaturedFilm