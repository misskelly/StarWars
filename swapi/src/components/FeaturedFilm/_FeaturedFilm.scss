.film-container {
  grid-area: film;
}

h3, h4, h5, {
  margin: 0;
}
.crawler {
  background: transparent;
}

.film-info {
  text-align: center;
}
.film-title {
  line-height: 1;
  font-family: $alien;
  margin-top: 3rem;
}