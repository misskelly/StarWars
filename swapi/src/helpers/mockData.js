export const mockCleanFilm = {
     scroll: "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy …",
     title: "The Phantom Menace",
     year: "1999"
   }
export const mockDirtyFilm = {
     opening_crawl: "Turmoil has engulfed the\r\nGalactic Republic. The taxation\r\nof trade routes to outlying star\r\nsystems is in dispute.\r\n\r\nHoping to resolve the matter\r\nwith a blockade of deadly\r\nbattleships, the greedy …",
     title: "The Phantom Menace",
     release_date: "1999-07-03"
   }


