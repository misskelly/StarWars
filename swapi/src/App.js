import React, { Component } from 'react';
// import { fetchFilm } from './helpers/fetchCalls.js'
import FeaturedFilm from './components/FeaturedFilm/FeaturedFilm'

export default class App extends Component {
  constructor() {
    super()
    this.state = {
      people: [],
      planets: [],
      vehicles: [],
      film: {},
      errorStatus: ''
    }
  }

  fetchFilm = () => {
    const randomNum = Math.floor(Math.random() * (7 - 1) + 1);
    const url = `https://www.swapi.co/api/films/${randomNum}`
    return fetch(url)
      .then(response => {
        if (response.status >= 400) {
          this.setState({errorStatus: 'Error fetching film info'});
        } else {
          response.json()
        .then(film => this.cleanFilm(film))
        .catch(error => console.log(error))
      }
    });
  }

  cleanFilm = film => {
    const { title, opening_crawl, release_date } = film
    const clean = { 
      title: title,
      scroll: opening_crawl,
      year: release_date.slice(0, 4)
    }
    this.setState({film: clean})
  }

  componentDidMount() {
    this.fetchFilm()
  }
  
  render() {
    const { film } = this.state;
    return (
      <div className="App">
    { /* Header */}
        <main>
          <h1>StarWars</h1>
        <FeaturedFilm film={ film }/>

        </main>
      </div>
  );
}

}

