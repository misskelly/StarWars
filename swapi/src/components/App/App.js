import React, { Component } from 'react';
import Gallery from '../Gallery/Gallery'
import Header from '../Header/Header'
import Loading from '../Loading/Loading'
import Home from '../Home/Home'
import { handleClean } from '../../helpers/cleaners'
import { fetchFilm, fetchData } from '../../helpers/fetchCalls.js'



export default class App extends Component {
  constructor() {
    super()
    this.state = {
      people: [],
      planets: [],
      vehicles: [],
      film: {},
      loading: false,
      errorStatus: '',
      activeComponent: ''
    }
    this.updateActiveComponent = this.updateActiveComponent.bind(this);
  }

  componentDidMount() {
    this.setState({loading: true})
    fetchFilm()
      .then(film => this.setState({ film, loading: false }))
    }
  
  getData(category) {
    const url = `https://www.swapi.co/api/${category}/`;
    return fetchData(url)
      .then(data => handleClean(data, category))
      .then(result => this.setState({ [category]: result, loading: false }))
      .catch(err => this.setState({ errorStatus: 'Error fetching data' }))
  }

  updateActiveComponent(category) {
    this.setState({ activeComponent: category, loading: true });
    this.getData(category);
  }

    
    render() {
      const { film, activeComponent, loading } = this.state;
      
    return (
      <main className='App'>
        <div className='stars'></div>
        <div className='twinkling'></div>
        <section className='container'>
        { loading === true && <Loading /> }
        {loading === false &&
          <Header updateActiveComponent={ this.updateActiveComponent }/>
        }
        { loading === false && activeComponent === '' &&
          <Home updateActiveComponent={ this.updateActiveComponent } film={ film } />
        }
          { activeComponent !== '' && loading === false &&
      
          <Gallery collection={this.state[activeComponent]} category={ activeComponent }/>
        }
        
        </section>
      </main>
    );
  }
}