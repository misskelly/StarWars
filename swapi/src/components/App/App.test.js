import React from 'react';
import App from './App';
import FeaturedFilm from '../FeaturedFilm/FeaturedFilm';
import { mount, shallow } from 'enzyme';
// import { fetchFilm as mockFetchFilm } from '../../helpers/fetchCalls.js'
import { mockCleanFilm, mockDirtyFilm } from '../../helpers/mockData.js';


describe('App', () => {
  let wrapper;
  let initialState;
  
  beforeEach(() => {
    wrapper = shallow(<App />);
    initialState = {
      people: [],
      planets: [],
      vehicles: [],
      film: {},
      errorStatus: '',
      activeComponent: ''
    }
  });
  
  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
  
  
 // const mockActiveComponent = 'planets'
 // const mockEvent = jest.fn()
 // const mockFetchFilm = jest.fn()
 // const mockFetchPlanets = jest.fn()
  // window.fetch = jest.fn()
  //   .mockImplementation(
  //     () => Promise.resolve(
  //       { json: () => Promise.resolve(mockDirtyFilm) }
  //       )
  //   )
  // })

  it('has the correct default state', () => {
    expect(wrapper.state()).toEqual(initialState);
  });

  it('renders a FeaturedFilm component with the correct props', () => {

    wrapper.setState({ film: mockCleanFilm});
    expect(wrapper.find(FeaturedFilm).prop('film')).toEqual(wrapper.state('film'));

  });

  // // on component did mount we call fetch film
  // it('should call fetchFilm on mount', () => {
  //   // mount component
  //   jest.mock('../../helpers/fetchcalls.js')
  //    expect(mockFetchFilm).toHaveBeenCalled();
  //     // mock out if fetch is called, what to return
      
  //     // assert that state was set with film
  //     // state updates with new film after cdm
  // })





  it('should call fetchPlanets on button click', () => {

  })

});
// on planets button click, we call fetch planets
// state updates with active component: planets