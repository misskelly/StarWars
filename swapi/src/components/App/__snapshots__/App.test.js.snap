// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<main
  className="App"
>
  <div
    className="stars"
  />
  <div
    className="twinkling"
  />
  <section
    className="home"
  >
    <Header />
    <section
      className="left-btns"
    >
      <button
        className="planets-btn"
        onClick={[Function]}
      >
        Planets
      </button>
      <button
        className="planets-btn"
        onClick={[Function]}
      >
        Planets
      </button>
    </section>
    <FeaturedFilm
      film={Object {}}
    />
    <section
      className="right-btns"
    >
      <button
        className="planets-btn"
        onClick={[Function]}
      >
        Planets
      </button>
      <button
        className="planets-btn"
        onClick={[Function]}
      >
        Planets
      </button>
    </section>
    <Ship />
  </section>
</main>
`;
